### DISCUSSION TITLE
Allow assignment of value from ui.number function to a variable

### DISCUSSION THREAD

**Discussion by AliMostafaRadwan** [zauberzeug/nicegui#859] [Ideas / Feature Requests]
Currently, when using the ui.number function in GitHub UI, it does not allow assigning the selected value to a variable directly. However, this feature is available in Streamlit, where users can assign the value from st.number_input to a variable. This GitHub suggestion proposes adding similar functionality to the ui.number function in order to improve usability and streamline the coding process.

Expected Behavior:
When using the ui.number function, users should be able to assign the selected value to a variable directly, similar to how it is done in Streamlit's `st.number_input`.

Example:

```python

# Current behavior
num = ui.number('Number', min=0, max=100, step=1, value=50, on_change=on_change)
print(num)  # Outputs None

# Proposed behavior
num = ui.number('Number', min=0, max=100, step=1, value=50, on_change=on_change)
print(num)  # Outputs the selected number

```
By allowing the value from ` ui.number`  to be assigned to a variable, it would enhance code readability and make it easier to use the selected value in subsequent calculations or operations. This improvement would align the functionality of GitHub UI more closely with other popular UI frameworks like Streamlit.

Please consider implementing this feature to provide a more intuitive and efficient user experience. Thank you!

**Comment by rodja** [comment]
We explicitly started NiceGUI because [we disliked the super strange (and non-Python) flow of Streamlit](https://github.com/zauberzeug/nicegui/discussions/21). Let's have a look at your example:

```py
num = ui.number('Number', min=0, max=100, step=1, value=50, on_change=on_change)
print(num)  # Outputs the selected number
```

To implement this with Python, `ui.number` must block until an input is performed. That means all code below `ui.number` is not executed. Streamlit works around that by re-executing all code on each input (which makes it slow and unusable if you want to integrate "normal" Python code). Blocking itself would also block all other server responses, because NiceGUI uses an event loop on the main thread.

But we could do it in a similar way as #849 without violating Python:

```py
@ui.page('/')
async def index():
    num = await ui.number().blured()
    ui.notify(num)
```
I picked `blured()` here to be consistent with [html vocabulary](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/blur).

If we implement this pattern for all elements, we could do something very similar to [PyWebIO](https://github.com/pywebio/PyWebIO/) where new elements appear after user input is made.
